pkg/azureutils/azure_disk_utils.go

```diff
@@ -453,3 +453,43 @@ func getKubeClient(kubeconfig string) (*kubernetes.Clientset, error) {
 
 	return kubernetes.NewForConfig(config)
 }
+
+// FindKeyValueInMapCaseInsensitive: find <key, value> pair in map
+func FindKeyValueInMapCaseInsensitive(targetMap map[string]*string, key, value string) bool {
+	for k, v := range targetMap {
+		if v != nil {
+			if strings.EqualFold(k, key) && strings.EqualFold(*v, value) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// InsertDiskProperties: insert disk properties to map
+func InsertDiskProperties(disk *compute.Disk, publishConext map[string]string) {
+	if disk == nil || publishConext == nil {
+		return
+	}
+
+	if disk.Sku != nil {
+		publishConext[consts.SkuNameField] = string(disk.Sku.Name)
+	}
+	prop := disk.DiskProperties
```

Comment:
Can we add location and maxShares properties as well to this list.

Reply:
even dynamic PV does not have `location` field, and `availabilityZone` is already in the `allowedTopologies` field of PV, same as dynamic PV.

added `maxShares`